### PR TITLE
feat(mvp_instructor): Added basic functionality for adding units, preview courselets

### PR DESCRIPTION
**Pull request by alexbojko**
Added basic functionality for adding units, templates, urls,  views, functionality for adding error models (formset, view).

branch: feature/mvp_instructor

fixes issues:
 - #312 
 - #309

### CODE REVIEW

File: mysite/ctms/views.py [hunk 1]
```diff
@@ -404,3 +403,212 @@ def get_context_data(self, **kwargs):
             'courslet': self.get_courslet()
         })
         return kwargs
+
+
+class FormSetMixin(object):
```
Comment (maxsocl) [inline mysite/ctms/views.py]: Wrong naming - this class is not looks like Mixin.

File: mysite/chat/fsm_plugin/chat_add_lesson.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    fsm = edge.fromNode.fsm
+    unitStatus = fsmStack.state.get_data_attr('unitStatus')
+    nextUL = unitStatus.get_lesson()
+    if nextUL.is_question():
+        return fsm.get_node(name='ASK')
+    else:  # just a lesson to read
+        return edge.toNode
+
+
+class START_(object):
```
Comment (alexbojko) [inline mysite/chat/fsm_plugin/chat_add_lesson.py]: we'll need these nodes in future.

File: mysite/chat/views.py [hunk 3]
```diff
@@ -68,46 +70,61 @@ def get_will_learn_need_know(self, unit, courseUnit):
                         contaner.add((title, url))
             return will_learn, need_to_know
 
+    def check_course_not_published(self, request, courseUnit):
```
Comment (alexbojko) [inline mysite/chat/views.py]: renamed to `check_course_not_published_and_user_is_not_instuctor`

File: mysite/chat/views.py [hunk 4]
```diff
@@ -68,46 +70,61 @@ def get_will_learn_need_know(self, unit, courseUnit):
...
                 'lti/error.html',
                 {'message': 'This Courselet is not published yet.'}
             )
-        if not Role.objects.filter(
-            user=request.user.id, course=courseUnit.course, role=Role.ENROLLED
-        ):
+        if not self.user_enrolled(request, courseUnit):
             enrolling = Role.objects.get_or_create(user=request.user,
                                                    course=courseUnit.course,
                                                    role=Role.SELFSTUDY)[0]
```
Comment (alexbojko) [inline mysite/chat/views.py]: I just refactored this part of code - moved some parts from function to outer functions to make it replaceable.

Comment (maxsocl) [inline mysite/chat/views.py]: ok

Comment (maxsocl) [inline mysite/chat/views.py]: Need to check this code aims.

**Pull request merged**